### PR TITLE
Scaladoc: Add necessary parentheses in function types

### PR DESCRIPTION
**Pull request by pikinier20**
closes: #14511

### CODE REVIEW

File: scaladoc-testcases/src/tests/functionTypeSignatures.scala [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package tests.functionTypeSignatures
+
+type A = ((Int, Int)) => Int
+
+type B = (Int | String) => Int
```
Comment (KacperFKorban) [inline scaladoc-testcases/src/tests/functionTypeSignatures.scala:5]: `|` and `&` type operators bind stronger than `=>` so it's a judgment call it they should be in parentheses.

**Review by KacperFKorban — APPROVED** [review]
Overall looks good to me

File: scaladoc-testcases/src/tests/functionTypeSignatures.scala (hunk 1 above)
Comment (pikinier20) [reply scaladoc-testcases/src/tests/functionTypeSignatures.scala:5]: I think for the sake of clarity we should keep them in parentheses

Comment (KacperFKorban) [reply scaladoc-testcases/src/tests/functionTypeSignatures.scala:5]: Ok, makes sense

**Pull request merged**